ecosystem/python/sdk/aptos_sdk/ed25519.py

```diff
@@ -92,6 +94,42 @@ def serialize(self, serializer: Serializer):
         serializer.to_bytes(self.key.encode())
 
 
+class MultiEd25519PublicKey:
+    keys: List[PublicKey]
+    threshold: int
+
+    def __init__(self, keys: List[PublicKey], threshold: int):
+        self.keys = keys
+        self.threshold = threshold
+
+    def __str__(self) -> str:
+        return f"{self.threshold}-of-{len(self.keys)} Multi-Ed25519 public key"
+
+    def auth_key(self) -> bytes:
+        hasher = hashlib.sha3_256()
+        hasher.update(self.to_bytes() + b"\x01")
+        return hasher.digest()
+
+    def to_bytes(self) -> bytes:
+        concatenated_keys = bytes()
+        for key in self.keys:
+            concatenated_keys += key.key.encode()
+        return concatenated_keys + bytes([self.threshold])
+
+    @staticmethod
+    def from_bytes(key: bytes) -> MultiEd25519PublicKey:
+        n_signers = int(len(key) / PublicKey.LENGTH)
+        keys = []
+        for i in range(n_signers):
+            start_byte = i * PublicKey.LENGTH
+            end_byte = (i + 1) * PublicKey.LENGTH
+            keys.append(PublicKey(VerifyKey(key[start_byte:end_byte])))
+        return MultiEd25519PublicKey(keys, int(key[-1]))
+
+    def serialize(self, serializer: Serializer):
```

Comment:
same here, we have no tests, we could probably generate some strings in the aptos rust code as samples

at least have some e2e tests that show that a set of public keys can verify a signature

Reply:
@davidiw e2e tests added.